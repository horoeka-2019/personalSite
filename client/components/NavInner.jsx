import React from "react"
import { Link, HashRouter as Router } from "react-router-dom"

class Nav extends React.Component {
  render() {
    return (
      <div>
        <ul className="navigation">
          <div className="nav-right">
              <Router>
            <li>
              <Link to='/' className="burger">Home</Link>
            </li>
            <li>
              <Link to='/about' className="burger">About</Link>
            </li>
            <li>
              <Link to='/work' className="burger">Work</Link>
            </li>
            <li>
              <Link to='/blog' className="burger">Blog</Link>
            </li>
            </Router>
          </div>
        </ul>
      </div>
    );
  }
}

export default Nav;
