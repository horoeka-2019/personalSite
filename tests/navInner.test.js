import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import NavInner from '../client/components/NavInner'

describe('<NavInner /> component test', () => {
  it('renders nav bar', () => {
    // Arrange
    const expected = 'HomeAboutWork'

    // Act
    const wrapper = mount(<NavInner />)
    const actual = wrapper.text()
    // Assert
    expect(actual).toContain(expected)
  })
  it('render unchanged from previous snapshot', () => {
    const wrapper = mount(<NavInner/>)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
 
})