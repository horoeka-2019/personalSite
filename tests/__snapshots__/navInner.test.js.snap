// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavInner /> component test render unchanged from previous snapshot 1`] = `
<Nav>
  <div>
    <ul
      className="navigation"
    >
      <div
        className="nav-right"
      >
        <HashRouter>
          <Router
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
          >
            <li>
              <NavLink
                activeClassName="navbar__link--"
                className="burger"
                to="/"
              >
                <Link
                  aria-current="page"
                  className="burger navbar__link--"
                  style={Object {}}
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current="page"
                    className="burger navbar__link--"
                    href="#/"
                    navigate={[Function]}
                    style={Object {}}
                  >
                    <a
                      aria-current="page"
                      className="burger navbar__link--"
                      href="#/"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      Home
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="navbar__link--active"
                className="burger"
                to="/about"
              >
                <Link
                  aria-current={null}
                  className="burger"
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/about",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current={null}
                    className="burger"
                    href="#/about"
                    navigate={[Function]}
                  >
                    <a
                      aria-current={null}
                      className="burger"
                      href="#/about"
                      onClick={[Function]}
                    >
                      About
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="navbar__link--active"
                className="burger"
                to="/work"
              >
                <Link
                  aria-current={null}
                  className="burger"
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/work",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current={null}
                    className="burger"
                    href="#/work"
                    navigate={[Function]}
                  >
                    <a
                      aria-current={null}
                      className="burger"
                      href="#/work"
                      onClick={[Function]}
                    >
                      Work
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
            </li>
          </Router>
        </HashRouter>
      </div>
    </ul>
  </div>
</Nav>
`;
